File: features/wx/ts/Ticket.ts
import { int64 } from "./lib/less";
import { UserType } from './UserType';

export enum TicketType {
    Jsapi = "jsapi",
    Card = "wx-card"
}

/**
 * Ticket
 * @type db
 */
export class Ticket {

    /**
     * ID
     */
    id: int64 = 0

    /**
     * 类型
     * @index ASC
     * @length 32
     */
    type: TicketType = TicketType.Jsapi

    /**
     * appid
     * @index ASC
     * @length 64
     */
    appid: string = ''

    /**
     * ticket
     * @length 255
     * @output false
     */
    ticket: string = ''

    /**
     * 过期时间
     */
    etime: int64 = 0

}

File: features/wx/wx/Ticket.go
package wx

import (
	"github.com/hailongz/golang/db"
)

type Ticket struct {
	db.Object
	Type	string	`json:"type" name:"type" title:"类型" length:"32" index:"ASC"`
	Appid	string	`json:"appid" name:"appid" title:"appid" length:"64" index:"ASC"`
	Ticket	string	`json:"-" name:"ticket" title:"ticket" length:"255"`
	Etime	int64	`json:"etime" name:"etime" title:"过期时间"`
}

func (O *Ticket) GetName() string {
	return "ticket"
}

func (O *Ticket) GetTitle() string {
	return "Ticket"
}

